streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -553,6 +553,7 @@ class StreamingContext private[streaming] (
    */
   def stop(stopSparkContext: Boolean = true): Unit = synchronized {
     stop(stopSparkContext, false)
+    this.uiTab.foreach(StreamingTab.detachStreamingTab(this, _))
```

Comment:
Why is it hear and not in the other `stop` The tab should be removed when the stremaing context is stopped, so both `stop` should detach the tab.